src/lib/mongoBackend/mongoUpdateContextSubscription.cpp

```diff
@@ -107,14 +111,84 @@ HttpStatusCode mongoUpdateContextSubscription
    */
   BSONObjBuilder newSub;
 
-  /* Entities, attribute list and reference are not updatable, so they are appended directly */
-  newSub.appendArray(CSUB_ENTITIES, getField(sub, CSUB_ENTITIES).Obj());
-  newSub.appendArray(CSUB_ATTRS, getField(sub, CSUB_ATTRS).Obj());
-  newSub.append(CSUB_REFERENCE, getStringField(sub, CSUB_REFERENCE));
+  if (version != "v2") {
+    /* Entities, attribute list and reference are not updatable, so they are appended directly */
+    newSub.appendArray(CSUB_ENTITIES, getField(sub, CSUB_ENTITIES).Obj());
+    newSub.appendArray(CSUB_ATTRS, getField(sub, CSUB_ATTRS).Obj());
+    newSub.append(CSUB_REFERENCE, getStringField(sub, CSUB_REFERENCE));
+  }
+  else // v2
+  {
+    // Reference
+    std::string ref;
+    if (!requestP->reference.isEmpty())
+    {
+      ref = requestP->reference.get();
+    }
+    else
+    {
+      ref = getStringField(sub, CSUB_REFERENCE);
```

Comment:
In a recent PR, we instroduced reference URL checking. Just to ensure that this PR keep that checking (maybe it is done at parsing stage?).